rootfs/etc/nginx/lua/lua_ingress.lua

```diff
@@ -2,25 +2,40 @@ local _M = {}
 
 local seeds = {}
 local original_randomseed = math.randomseed
-math.randomseed = function(seed)
-  local pid = ngx.worker.pid()
 
-  if seeds[pid] then
+local function get_seed_from_urandom()
+  local seed
+  local frandom = io.open("/dev/urandom", "rb")
+  if frandom then
+    local str = frandom:read(4)
+    frandom:close()
+    seed = 0
+    for i = 1, 4 do
+        seed = 256 * seed + str:byte(i)
+    end
+  end
+  return seed
+end
+
+math.randomseed = function()
+  local pid = ngx.worker.pid()
+  local seed = seeds[pid]
+  if seed then
     ngx.log(ngx.WARN,
-      string.format("ignoring math.randomseed(%d) since PRNG is already seeded for worker %d", seed, pid))
+      string.format("ignoring math.randomseed() since PRNG is already seeded for worker %d", pid))
     return
   end
 
+  seed = get_seed_from_urandom()
+  if not seed then
+    seed = ngx.now() * 1000 + pid
```

Comment:
Why not `ngx.time()`? That way you'd not have to multiply by 1000.

Reply:
`ngx.time()` returns seconds, and `ngx.now()` returns seconds and milliseconds, so `ngx.time() + pid` maybe get the same seed, however `ngx.now() * 1000 + pid` is less chance to the same seed.